src/herder/Upgrades.cpp

```diff
@@ -403,6 +552,25 @@ Upgrades::isValidForApply(UpgradeType const& opaqueUpgrade,
                                         ProtocolVersion::V_18) &&
               (upgrade.newFlags() & ~MASK_LEDGER_HEADER_FLAGS) == 0;
         break;
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    case LEDGER_UPGRADE_CONFIG:
+    {
+        auto cfgUpgrade =
+            ConfigUpgradeSetFrame::makeFromKey(ltx, upgrade.newConfig());
+        if (!cfgUpgrade)
+        {
+            return UpgradeValidity::XDR_INVALID;
```

Comment:
we should not return `XDR_INVALID` here as this error is reserved for when the actual upgrade is bad. Here we'd return this code for all sorts of other reasons, including if the upgrade references missing ledger entries.

Reply:
Updated to return `INVALID`.